hedera-node/test-clients/src/main/java/com/hedera/services/bdd/spec/transactions/node/HapiNodeCreate.java

```diff
@@ -0,0 +1,179 @@
+        return this;
+    }
+
+    public HapiNodeCreate accountId(final AccountID accountID) {
+        this.accountId = Optional.of(accountID);
+        return this;
+    }
+
+    public HapiNodeCreate description(final String description) {
+        this.description = Optional.of(description);
+        return this;
+    }
+
+    public HapiNodeCreate gossipEndpoint(final List<ServiceEndpoint> gossipEndpoint) {
+        this.gossipEndpoints = gossipEndpoint;
+        return this;
+    }
+
+    public HapiNodeCreate serviceEndpoint(final List<ServiceEndpoint> serviceEndpoint) {
+        this.serviceEndpoints = serviceEndpoint;
+        return this;
+    }
+
+    public HapiNodeCreate gossipCaCertificate(final byte[] gossipCaCertificate) {
+        this.gossipCaCertificate = Optional.of(gossipCaCertificate);
+        return this;
+    }
+
+    public HapiNodeCreate grpcCertificateHash(final byte[] grpcCertificateHash) {
+        this.grpcCertificateHash = Optional.of(grpcCertificateHash);
+        return this;
+    }
+
+    @Override
+    protected HapiNodeCreate self() {
+        return this;
+    }
+
+    @Override
+    protected long feeFor(@NonNull final HapiSpec spec, @NonNull final Transaction txn, final int numPayerKeys)
+            throws Throwable {
+        return spec.fees().forActivityBasedOp(HederaFunctionality.CryptoCreate, this::usageEstimate, txn, numPayerKeys);
+    }
+
+    private FeeData usageEstimate(final TransactionBody txn, final SigValueObj svo) {
+        // temp till we decide about the logic
+        return FeeData.newBuilder()
+                .setNodedata(FeeComponents.newBuilder().setBpr(0))
+                .setNetworkdata(FeeComponents.newBuilder().setBpr(0))
+                .setServicedata(FeeComponents.newBuilder().setBpr(0))
+                .build();
+    }
+
+    @Override
+    protected Consumer<TransactionBody.Builder> opBodyDef(@NonNull final HapiSpec spec) throws Throwable {
+        final NodeCreateTransactionBody opBody = spec.txns()
+                .<NodeCreateTransactionBody, NodeCreateTransactionBody.Builder>body(
+                        NodeCreateTransactionBody.class, builder -> {
+                            accountId.ifPresent(builder::setAccountId);
+                            description.ifPresent(builder::setDescription);
+                            builder.addAllGossipEndpoint(gossipEndpoints);
+                            builder.addAllServiceEndpoint(serviceEndpoints);
+                            gossipCaCertificate.ifPresent(s -> builder.setGossipCaCertificate(ByteString.copyFrom(s)));
+                            grpcCertificateHash.ifPresent(s -> builder.setGrpcCertificateHash(ByteString.copyFrom(s)));
+                        });
+        return b -> b.setNodeCreate(opBody);
+    }
+
+    @Override
+    protected List<Function<HapiSpec, Key>> defaultSigners() {
+        return Arrays.asList(spec -> spec.registry().getKey(effectivePayer(spec)));
+    }
```

Comment:
This needs to include adminKey also.

Reply:
This will be addressed in a separate PR. Added TODO comment.